src/utils/provider.ts

```diff
@@ -76,3 +76,26 @@ export function makeExplorerUrl(
       console.warn(`No explorer for chain ID ${chainId}; tx Hash: ` + txHash);
   }
 }
+
+export async function switchNetwork(
+  targetChainId: string,
+  onError?: (e: Error | any) => void
+): Promise<void> {
+  const ethereum = (window as any).ethereum;
+  if (!ethereum) {
+    return;
+  }
+  // convert decimal string to hex
+  const targetChainIdHex = '0x' + parseInt(targetChainId).toString(16);
+
+  try {
+    await ethereum.request({
+      method: 'wallet_switchEthereumChain',
+      params: [{ chainId: targetChainIdHex }],
+    });
+    // refresh
+    (window as any).location.reload();
```

Comment:
any way to prevent this if the user declines the request?

Reply:
like prevent this reload callback binding to the page?

I think this is setup even before the user signs any request, so not sure what functionaltiy you are thinking of?

Reply:
Requesting to switch the chain shows a prompt. Can we have the page not reload if the user chooses to decline the prompt?

Reply:
Never mind--it already does this. Must have gotten mixed up when I tested earlier.